hypex/dataset/datagenerator.py

```diff
@@ -0,0 +1,293 @@
+        self.num_discrete_features = num_discrete_features
+        self.is_binary_target = is_binary_target
+        self.na_columns = na_columns
+        self.na_step = na_step
+        self.df: pd.DataFrame = pd.DataFrame()
+        self.treatment_name: Union[str, list, None] = None
+        self.target_names: list = []
+        self.info_col_names: Union[list, None] = None
+        self.features_names: list = []
+        self.ate: float = 0.0
+        self.sigma = 1.3
+        self._generate_df()
+
+    def _generate_df(self):
+        """
+        Generates the dataset and predict ate for it
+        """
+        m_with_dummy = self._generate_feature_cols()
+        info = self._generate_info_cols()
+        treatment = self._generate_treatment_cols()
+        data = self._generate_outcome_and_full_data(m_with_dummy, treatment)
+        self._set_column_names()
+        self.df = self._set_df(data, info)
+        self._set_nans()
+        self._process_ate()
+
+    def _process_ate(self):
+        pass
+
+    def _generate_feature_cols(self):
+        """
+        Generate columns with main features for causal process
+
+        Returns:
+            m_with_dummy:
+                Array with features
+        """
+        means = np.random.uniform(-1, 1, self.num_features)
+        cov_mat = np.identity(self.num_features)
+        m = np.random.multivariate_normal(means, cov_mat, self.num_records)
+        m_with_dummy = to_categories(m, self.num_features, self.num_discrete_features)
+        return m_with_dummy
+
+    def _generate_info_cols(self):
+        """
+        Generate columns with information features for causal process
+
+        Returns:
+            info:
+                List with features
+        """
+        info = []
+        for i in range(self.num_info_cols):
+            if i % 2 == 0:
+                coefficient = 3 if i <= 2 else i
+                array = list(range(1, self.num_records * coefficient, coefficient))
+                random.shuffle(array)
+                info.append(array)
+            else:
+                unique_letters = list(set(string.ascii_letters.upper()))
+                letters = [random.choice(unique_letters) for _ in range(i * 2)]
+                info.append([letters[j] * ((i // 2) + 1) for j in np.random.choice(a=np.arange(i * 2),
+                                                                                   size=self.num_records)])
+        return info
+
+    def _generate_treatment_cols(self):
+        """
+        Generate columns with treatment values
+
+        Returns:
+            treatment:
+                Array with treatment
+        """
+        if self.is_treatment and self.num_targets == 1:
+            treatment = np.random.normal(0, self.sigma, (self.num_records, 1))
+        else:
+            treatment = np.random.normal(0, self.sigma, (self.num_records, self.num_targets))
+        treatment = np.digitize(treatment, bins=np.array([0, 0.5, 1])).astype(bool)
```

Comment:
couldn't understand logic here, please help me. 
Here we will have 3 treatments type [0, 1, 2] ? 

How does num_targets affect the number of treatments? 
Ideally, we should have one column with the impact. Only in the case of matching it will be [0, 1], and in the case of AB-testing, it can be [0, 1,..., n].

Reply:
I'm also not sure about using the normal distribution to generate treatment. In n case, we will see a clear imbalance of classes. 

Can we do it through some kind of partition coefficient? A parameter to split the treatment by? We'll have to think about it

Reply:
U.P.D. astype(bool), it would be [0, 1], but I'm not sure if this distribution is correct.